packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -472,6 +504,71 @@ export class IamPolicyChecksWebview extends VueWebview {
         }
     }
 
+    public async checkNoPublicAccess(documentType: PolicyChecksDocumentType, cfnParameterPath?: string) {
+        const document = IamPolicyChecksWebview.editedDocumentFileName
+        customPolicyCheckDiagnosticCollection.clear()
+
+        switch (documentType) {
+            case 'Terraform Plan': {
+                if (isTerraformPlan(document)) {
+                    const command = 'tf-policy-validator'
+                    const args = [
+                        'check-no-public-access',
+                        '--template-path',
+                        `${document}`,
+                        '--region',
+                        `${this.region}`,
+                        '--config',
+                        `${globals.context.asAbsolutePath(defaultTerraformConfigPath)}`,
+                    ]
+                    this.executeCustomPolicyChecksCommand({
+                        command,
+                        args,
+                        cfnParameterPathExists: !!cfnParameterPath,
+                        documentType,
+                        checkType: 'CheckNoPublicAccess',
+                    })
+                    return
+                } else {
+                    this.onCustomPolicyCheckResponse.fire([
+                        IamPolicyChecksConstants.IncorrectFileExtension,
+                        getResultCssColor('Error'),
+                    ])
+                    return
+                }
+            }
+            case 'CloudFormation': {
+                if (isCloudFormationTemplate(document)) {
+                    const command = 'cfn-policy-validator'
+                    const args = [
+                        'check-no-public-access',
+                        '--template-path',
+                        `${document}`,
+                        '--region',
+                        `${this.region}`,
+                    ]
+                    if (cfnParameterPath !== '') {
+                        args.push('--template-configuration-file', `${cfnParameterPath}`)
+                    }
+                    this.executeCustomPolicyChecksCommand({
```

Comment:
off topic for this PR, but important: I see that `executeCustomPolicyChecksCommand` does not use ChildProcess https://github.com/aws/aws-toolkit-vscode/blob/8ee7242fc443b37d622b59b5f19f7a086f3cf05d/packages/core/src/shared/utilities/childProcess.ts#L68

It's strongly recommended to use that if possible, because it provides improved logging and features, and in the future it will gain performance monitoring and telemetry enhancements.

Reply:
Thank you for pointing this out, I'll create a ticket for our team to investigate refactoring to use ChildProcess